### ISSUE TITLE
 tensorflow/core/grappler/costs/op_level_cost_estimator.cc:654] Check failed: iz % kz == 0 (256 vs. 0)Input channel 256 is not a multiple of filter channel 512.

### ISSUE THREAD

**Issue by Wmir** [tensorflow/tensorflow#49112]
I am implementing the following code i.e CBAM module in a network, however, I am getting the **following error:**

F tensorflow/core/grappler/costs/op_level_cost_estimator.cc:654] Check failed: iz % kz == 0 (256 vs. 0)Input channel 256 is not a multiple of filter channel 512

Here is the code of the CBAM block

`import tensorflow as tf

def CBAM(input, reduction):
    """
    @Convolutional Block Attention Module
    """

    _, channel, width, height = input.get_shape()  # (B, W, H, C)

    input = tf.transpose(input, perm=[0, 3, 2, 1]) 
    # channel attention
    x_mean = tf.reduce_mean(input, axis=(1, 2), keepdims=True)   # (B, 1, 1, C)
    x_mean = tf.layers.conv2d(x_mean, channel // reduction, 1, activation=tf.nn.relu, name='CA1', reuse=tf.AUTO_REUSE)  # (B, 1, 1, C // r)
    x_mean = tf.layers.conv2d(x_mean, channel, 1, name='CA2', reuse=tf.AUTO_REUSE)   # (B, 1, 1, C)

    x_max = tf.reduce_max(input, axis=(1, 2), keepdims=True)  # (B, 1, 1, C)
    x_max = tf.layers.conv2d(x_max, channel // reduction, 1, activation=tf.nn.relu, name='CA1', reuse=tf.AUTO_REUSE)
    # (B, 1, 1, C // r)
    x_max = tf.layers.conv2d(x_max, channel, 1, name='CA2', reuse=tf.AUTO_REUSE)  # (B, 1, 1, C)

    x = tf.add(x_mean, x_max)   # (B, 1, 1, C)
    x = tf.nn.sigmoid(x)        # (B, 1, 1, C)
    x = tf.multiply(input, x)   # (B, W, H, C)

    # spatial attention
    y_mean = tf.reduce_mean(x, axis=3, keepdims=True)  # (B, W, H, 1)
    y_max = tf.reduce_max(x, axis=3, keepdims=True)  # (B, W, H, 1)
    y = tf.concat([y_mean, y_max], axis=-1)     # (B, W, H, 2)
    y = tf.layers.conv2d(y, 1, 7, padding='same', activation=tf.nn.sigmoid)    # (B, W, H, 1)
    y = tf.multiply(x, y)  # (B, W, H, C)

    y = tf.transpose(y, perm=[0, 3, 2, 1])
    return y`

**I have called CBAM module in-network as given below:**

`layer17 = tf.layers.conv2d(layer16, 512, 3,
                           padding="same",
                           activation=tf.nn.relu,
                           dilation_rate=2,
                           data_format=self._data_format,
                           name="conv5/conv5_3")

layer17=CBAM(layer17, 8)
layer18 = tf.layers.max_pooling2d(layer17, 2, 1,
                                  padding="same",
                                  data_format=self._data_format)`

**Comment by UsharaniPagadala** [comment]
@Wmir 

We see that the issue template has not been filled, could you please do so as it helps us analyse the issue [tf version,stand alone code/colab gist to reproduce the issue faced].Thanks

**Comment by Wmir** [comment]
hi, UsharaniPagadala can you please elaborate a little bit more which template I need to fill? In the template section, i was asked three things that were not helpful for my scenario.

**Comment by UsharaniPagadala** [comment]
@Wmir 
Could you please reopen the issue with the clear details and provide a colab gist along with the dependencies to analyse the issue reported if the issue still persist.Thanks
